week1/Q2.py

```diff
@@ -0,0 +1,51 @@
+class Node:
```

Comment:
Adding comments to the classes and methods might help. They can be brief. You do not need to comment the meaning of self.